### DISCUSSION TITLE
Team 1736 Task Reports

### DISCUSSION THREAD

**Discussion by gerth2** [wpilibsuite/2024Beta#31] [Task Reports]
Progress Reporting for FRC team 1736 evaluating the 2024 beta

**Comment by gerth2** [comment]
Task 1 - Install

Win 11, installing over a fully-functional 2023 install (including robotpy)

Success on 2x pc's so far.

Overall good, but two notes:

NI installer still has this option as "optional" - I checked the box, but I'm not sure if it matters

![image](https://github.com/wpilibsuite/2024Beta/assets/4583662/d60b51a0-7ad3-41b6-94c3-14cfbeb02f5d)

Since we're running python we're not tied into using the wpilib vsCode install.... so I'd prefer to use our own. I had to click through the warnings that said "No, don't do this!". I knew enough to know it would be OK. Forward-going.... will the vsCode for wpilib include python support?

**Reply by calcmogul** [reply]
Yes.

**Comment by gerth2** [comment]
Task 2 - RIO Config

No issues observed! Got a RIO2 freshly imaged and configured via all the same processes we've used in past years. Docs appear accurate.

One minor note I went out of order: RobotPy is able to deploy to a roboRIO even without me configuring a team number, and the driver station did not recognize the RIO existed.

![image](https://github.com/wpilibsuite/2024Beta/assets/4583662/c776b65b-099d-4bf0-946b-4cf8eaa76dc6)

I think this is a corner case specific to my setup though. I had deployed to this same RIO with the 2023 setup, and my vsCode project... maybe... cached the IP address to use?

**Reply by virtuald** [reply]
Yes, there's a file that the robotpy-installer stores the team number in, so if you ran it from the same directory it would reuse that information.

**Comment by gerth2** [comment]
Task 3: Porting

Built, re-ran unit sims, deployed. Everything is passing.

[Starting from this python codebase developed over the summer with 2023](https://github.com/RobotCasserole1736/firstRoboPy/tree/v0.2). Zero changes required in core wpilib libraries.

The only thing required was commenting out pathplanner things, since these aren't yet supported.

**Reply by gerth2** [reply]
Update: This work has been tested on a real robot, no issues observed!

**Comment by gerth2** [comment]
Task 4: New Features

SysID - launches, and I can see where to put in my json file.

[Docs](https://docs.wpilib.org/en/latest/docs/software/pathplanning/system-identification/identification-routine.html) don't look up to date yet, I'm not sure what format to provide the .json in. I do have plenty of .wpilogs from on robot though, I can go get some and smush them into json format when needed.
